### ISSUE TITLE
s3-geotiff conflicts with geofence

### ISSUE THREAD

**Issue by fvanderbiest** [georchestra/georchestra#2225]
When creating a new s3-geotiff datastore:
```
Caused by: net.sf.ehcache.CacheException: Another unnamed CacheManager already exists in the same VM. Please provide unique names for each CacheManager in the config or do one of following:
1. Use one of the CacheManager.create() static factory methods to reuse same CacheManager with same name or create one if necessary
2. Shutdown the earlier cacheManager before creating new one with same name.
The source of the existing CacheManager is: URLConfigurationSource [url=jar:file:/var/lib/jetty/webapps/geoserver/WEB-INF/lib/geofence-persistence-3.2-20180503.181323-25.jar!/geofence-ehcache.xml]
	at net.sf.ehcache.CacheManager.assertNoCacheManagerExistsWithSameName(CacheManager.java:626)
	at net.sf.ehcache.CacheManager.init(CacheManager.java:391)
	at net.sf.ehcache.CacheManager.<init>(CacheManager.java:269)
	at org.geotools.s3.cache.CacheManagement.buildCache(CacheManagement.java:62)
	at org.geotools.s3.cache.CacheManagement.<init>(CacheManagement.java:38)
	at org.geotools.s3.cache.CacheManagement.<clinit>(CacheManagement.java:29)
	... 158 more
```

**Comment by fvanderbiest** [comment]
Tested with s3://landsat-pds/L8/001/002/LC80010022016230LGN00/LC80010022016230LGN00_B1.TIF?useAnon=true&awsRegion=US_WEST_2

**Comment by groldan** [comment]
Added `s3-geotiff` to the geoserver docker builds:
```
diff
diff --git a/Makefile b/Makefile
index 5ba19dcc5..258cda2c0 100644
--- a/Makefile
+++ b/Makefile
@@ -22,16 +22,16 @@ docker-build-gn3: docker-pull-jetty
 docker-build-geoserver: docker-pull-jetty
        cd geoserver; \
        rm -rf geoserver-submodule/data/citewfs-1.1/workspaces/sf/sf/E*; \
-       LANG=C mvn clean install -DskipTests; \
+       LANG=C mvn clean install -DskipTests -Ps3-geotiff; \
        cd webapp; \
-       mvn clean install docker:build -Pdocker,colormap,mbtiles,wps-download,app-schema,control-flow,csw,feature-pregeneralized,gdal,importer,inspire,libjpeg-turbo,monitor,pyramid,wps -DskipTests
+       mvn clean install docker:build -Pdocker,colormap,mbtiles,wps-download,app-schema,control-flow,csw,feature-pregeneralized,gdal,importer,inspire,libjpeg-turbo,monitor,pyramid,wps,s3-geotiff -DskipTests
 
 docker-build-geoserver-geofence: docker-pull-jetty
        cd geoserver; \
        rm -fr geoserver-submodule/data/citewfs-1.1/workspaces/sf/sf/E*; \
-       LANG=C mvn clean install -Pgeofence -DskipTests; \
+       LANG=C mvn clean install -Pgeofence,s3-geotiff -DskipTests; \
        cd webapp; \
-       mvn clean install docker:build -Pdocker,colormap,mbtiles,wps-download,app-schema,control-flow,csw,feature-pregeneralized,gdal,importer,inspire,libjpeg-turbo,monitor,pyramid,wps,geofence -DskipTests
+       mvn clean install docker:build -Pdocker,colormap,mbtiles,wps-download,app-schema,control-flow,csw,feature-pregeneralized,gdal,importer,inspire,libjpeg-turbo,monitor,pyramid,wps,geofence,s3-geotiff -DskipTests
 
 docker-build-console: build-deps docker-pull-jetty
        mvn clean package docker:build -Pdocker -DskipTests --pl console
Submodule geoserver/geoserver-submodule contains modified content
```
Works with `docker-compoese` using `georchestra/geoserver:latest`:
![image](https://user-images.githubusercontent.com/207423/53883055-bef8ec80-3ff6-11e9-8628-97ba3d552658.png)

**Comment by groldan** [comment]
Can't run the `geofence` enabled one, getting ```org.hibernate.MappingException: Could not determine type for: java.util.Set, at table: gf_layer_details, for columns: [org.hibernate.mapping.Column(styleName)]```

**Comment by severo** [comment]
See https://github.com/georchestra/georchestra/pull/2536

**Comment by groldan** [comment]
Vanilla geoserver master with s3-geotiff and geofence profiles work, at least I'm able to set up and use the s3 geotiff layer. I'm guessing I'm missing enabling caching on geofence or something like that. @fvanderbiest if you have steps to reproduce that'd be appreciated, I'm sure I'm missing something

**Comment by groldan** [comment]
ok, with #2536 applied:
```
<?xml version="1.0" encoding="UTF-8" standalone="no"?><!DOCTYPE ServiceExceptionReport SYSTEM "https://georchestra.mydomain.org/geoserver/schemas/wms/1.1.1/WMS_exception_1_1_1.dtd"> <ServiceExceptionReport version="1.1.1" >   <ServiceException>
      java.lang.ExceptionInInitializerErrorAnother unnamed CacheManager already exists in the same VM. Please provide unique names for each CacheManager in the config or do one of following:
1. Use one of the CacheManager.create() static factory methods to reuse same CacheManager with same name or create one if necessary
2. Shutdown the earlier cacheManager before creating new one with same name.
The source of the existing CacheManager is: URLConfigurationSource [url=jar:file:/tmp/jetty/jetty-0.0.0.0-8080-geoserver.war-_geoserver-any-1991734768024208485.dir/webapp/WEB-INF/lib/geofence-persistence-3.2-20180503.181323-25.jar!/geofence-ehcache.xml]
</ServiceException></ServiceExceptionReport>
```

**Comment by groldan** [comment]
Also getting this when trying to configure a new `s3-geotiff` store:
```
Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.geotools.s3.cache.CacheManagement
	at org.geotools.s3.S3ImageInputStreamImpl.<init>(S3ImageInputStreamImpl.java:75)
	at org.geotools.s3.geotiff.S3GeoTiffFormat.getReader(S3GeoTiffFormat.java:88)
	at org.geotools.s3.geotiff.S3GeoTiffFormat.getReader(S3GeoTiffFormat.java:38)
	at org.geoserver.catalog.ResourcePool.getGridCoverageReader(ResourcePool.java:1522)
	at org.geoserver.catalog.ResourcePool.getGridCoverageReader(ResourcePool.java:1468)
	at org.geoserver.catalog.impl.CoverageStoreInfoImpl.getGridCoverageReader(CoverageStoreInfoImpl.java:53)
```
`geofence-persistence-3.2-SNAPSHOT` depends on `ehcache:jar:1.5.0`, `gt-s3-geotiff-18-SNAPSHOT` depends on `ehcache:jar:2.10.3`, hence maven resolves to the latest one (`ehcache:jar:2.10.3`) for the geoserver built with both profiles.

**Comment by groldan** [comment]
Bye looking at the geotools s3-getotiff source code, it seems we can just have a separate ehcache config file and pass it through the `s3.caching.ehCacheConfig` system property, instead of relying on it setting up a default cache.

**Comment by groldan** [comment]
Fixed by #2543
